src/main/java/org/jabref/gui/externalfiles/FileFilterUtils.java

```diff
@@ -0,0 +1,100 @@
+package org.jabref.gui.externalfiles;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileTime;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FileFilterUtils {
+
+    public static LocalDateTime getFileTime(Path path) {
+        FileTime lastEditedTime = null;
+        try {
+            lastEditedTime = Files.getLastModifiedTime(path);
+        } catch (IOException e) {
+            System.err.println("Could not retrieve file time");
+        }
+        LocalDateTime localDateTime = lastEditedTime
+                .toInstant()
+                .atZone(ZoneId.systemDefault())
+                .toLocalDateTime();
+        return localDateTime;
+    }
+
+    public boolean isDuringLastDay(LocalDateTime fileEditTime) {
+        LocalDateTime NOW = LocalDateTime.now(ZoneId.systemDefault());
+        return fileEditTime.isAfter(NOW.minusHours(24));
+    }
+
+    public boolean isDuringLastWeek(LocalDateTime fileEditTime) {
+        LocalDateTime NOW = LocalDateTime.now(ZoneId.systemDefault());
+        return fileEditTime.isAfter(NOW.minusDays(7));
+    }
+
+    public boolean isDuringLastMonth(LocalDateTime fileEditTime) {
+        LocalDateTime NOW = LocalDateTime.now(ZoneId.systemDefault());
+        return fileEditTime.isAfter(NOW.minusMonths(1));
+    }
+
+    public boolean isDuringLastYear(LocalDateTime fileEditTime) {
+        LocalDateTime NOW = LocalDateTime.now(ZoneId.systemDefault());
+        return fileEditTime.isAfter(NOW.minusYears(1));
+    }
+
+    public static boolean filterByDate(Path path, String filter) {
+        FileFilterUtils fileFilter = new FileFilterUtils();
+        LocalDateTime fileTime = FileFilterUtils.getFileTime(path);
+        switch (filter) {
+            case "Last day":
+                return fileFilter.isDuringLastDay(fileTime);
+            case "Last week":
+                return fileFilter.isDuringLastWeek(fileTime);
+            case "Last month":
+                return fileFilter.isDuringLastMonth(fileTime);
+            case "Last year":
+                return fileFilter.isDuringLastYear(fileTime);
+            case "All time":
+                return true;
+            default:
+                return true;
+        }
+    }
+
+    public List<Path> sortByDateAscending(List<Path> files) {
+        return files.stream()
+                .sorted(Comparator.comparingLong(x -> FileFilterUtils.getFileTime(x)
+                        .atZone(ZoneId.systemDefault())
+                        .toInstant()
+                        .toEpochMilli()))
+                .collect(Collectors.toList());
+    }
+
+    public List<Path> sortByDateDescending(List<Path> files) {
+        return files.stream()
+                .sorted(Comparator.comparingLong(x -> -FileFilterUtils.getFileTime(x)
```

Comment:
Hello @Siedlerchr, thanks for the feedback! I think on this example we cannot replace the lambda with a method reference because from the method getFileTime the atZone method is called.
Thanks in advance.

![Screenshot_452](https://user-images.githubusercontent.com/56369797/123448999-efa82a00-d5e3-11eb-8a66-fd7b304d27fb.png)

Reply:
Ah okay, thanks, seems I overlooked that.  Then just please use a more meaningful name for the lambda parameter